tests/ops/models/test_consent_request.py

```diff
@@ -64,6 +58,40 @@ def test_consent(db):
     assert Consent.get(db, object_id=consent_2.id) is None
 
 
+def test_consent_with_gpc(db):
+    provided_identity_data = {
+        "privacy_request_id": None,
+        "field_name": "email",
+        "encrypted_value": {"value": "test@email.com"},
+    }
+    provided_identity = ProvidedIdentity.create(db, data=provided_identity_data)
+
+    consent_data_1 = {
+        "provided_identity_id": provided_identity.id,
+        "data_use": "user.biometric_health",
+        "opt_in": True,
+        "has_gpc_flag": True,
+        "conflicts_with_gpc": True,
+    }
+    consent_1 = Consent.create(db, data=consent_data_1)
+
+    consent_data_2 = {
+        "provided_identity_id": provided_identity.id,
+        "data_use": "user.browsing_history",
+        "opt_in": False,
+    }
+    consent_2 = Consent.create(db, data=consent_data_2)
+    data_uses = [x.data_use for x in provided_identity.consent]
+
+    assert consent_data_1["data_use"] in data_uses
+    assert consent_data_2["data_use"] in data_uses
+
+    provided_identity.delete(db)
+
+    assert Consent.get(db, object_id=consent_1.id) is None
+    assert Consent.get(db, object_id=consent_2.id) is None
```

Comment:
Is this to test cascading deletes? If yes all good, if it is just to clear the records there is a fixture that will clear them.

Reply:
Yes this tests the cascade delete from `ProvidedIdentity` to associated  `Consent` records